src/compiler/checker.ts

```diff
@@ -42580,6 +42581,17 @@ namespace ts {
 
                     if (name.kind === SyntaxKind.PropertyAccessExpression) {
                         checkPropertyAccessExpression(name, CheckMode.Normal);
+                        if (!links.resolvedSymbol) {
+                            const expressionType = checkExpressionCached(name.expression);
+                            const infos = getApplicableIndexInfos(expressionType, getLiteralTypeFromPropertyName(name.name));
+                            if (length(infos) && infos[0].declaration && infos[0].declaration?.symbol.flags & SymbolFlags.Signature && infos[0].declaration?.jsDoc) {

```

Comment:
What if `infos.length > 1` and `infos[0]` doesn’t have any JSDoc but `infos[1]` does? Why only check the first?

Reply:
not applicable anymore, since the conditions for the if statement changed.